index.js

```diff
@@ -64,10 +66,8 @@ module.exports = {
     ].filter(Boolean));
   },
 
-  treeForPublic() {
-    let parentAddon = this.parent.findAddonByName(this.parent.name());
-    if (!parentAddon) { return; }
-
+  treeForPublic() { 
+    let parentAddon = this.parent.treePaths ? this.parent : this;
```

Comment:
You're absolutely right. This is more related to when pulling `ember-cli-addon-docs` into your project, you get a `this.parent.name` is not a function and the project wouldn't run. 

```
this.parent.name is not a function
TypeError: this.parent.name is not a function
    at Class.treeForPublic (/Users/munsie/Documents/repos/testing-ui/node_modules/ember-cli-addon-docs/index.js:68:63)
    at Class._treeFor (/Users/munsie/Documents/repos/testing-ui/node_modules/ember-cli/lib/models/addon.js:551:33)
    at Class.treeFor (/Users/munsie/Documents/repos/testing-ui/node_modules/ember-cli/lib/models/addon.js:511:21)
    at addons.map.addon (/Users/munsie/Documents/repos/testing-ui/node_modules/ember-cli/lib/models/addon.js:383:34)
    at Array.map (native)
    at Class.eachAddonInvoke (/Users/munsie/Documents/repos/testing-ui/node_modules/ember-cli/lib/models/addon.js:381:24)
    at Class.treeFor (/Users/munsie/Documents/repos/testing-ui/node_modules/ember-cli/lib/models/addon.js:510:22)
    at project.addons.map.addon (/Users/munsie/Documents/repos/testing-ui/node_modules/ember-cli/lib/broccoli/ember-app.js:557:22)
    at Array.map (native)
    at EmberAddon.addonTreesFor (/Users/munsie/Documents/repos/testing-ui/node_modules/ember-cli/lib/broccoli/ember-app.js:555:32)
```